students/Ruohan/lesson03/mailroom.py

```diff
@@ -0,0 +1,95 @@
+#! /usr/bin/env python3
+
+''' simple mailroom proram for a non profit'''
+
+import sys
+
+datas = [['William Gates III',  653784.39, 2],
+         ['Mark Zuckerberg', 16396.10, 3],
+         ['Jeff Bezos', 877.33, 1],
+         ['Paul Allen', 708.42, 3]]
+
+'''make a donors' name list'''
+donors = []
+for data in datas:
+    donors.append(data[0])
+
+'''find if donor is already in the datas, if not add the name to the datas'''
+def find_donor(donor):
+    donor = donor.rstrip().title()
+    if donor in donors:
+        return donor
+    else:
+        datas.append([donor, 0, 0])
```

Comment:
What would happen `find_donor("Steve Ballmer")` is called once, his donation amount was increased, and `find_donor("Steve Ballmer")` is called again?

Reply:
the 'find_donor' function only used for find if the donor is already in the datas. If he is not, add him to the datas. Then use 'upgrade_dates' fuction to change the donation amount.

Reply:
But aren't you searching the `donors` list, not the `datas` list, and the `donors` list won't get any newly added donor?

Reply:
Oh.  I see. I will fix this problem!  Thanks Dr. Song!